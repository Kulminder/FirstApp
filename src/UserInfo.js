import React from 'react';
import PropTypes from 'prop-types';

const UserInfo = (props) => (
  <li className="person-list">
    <img alt={props.user.first_name} className="avatar-img" src={props.user.avatar} />
    <div className="person-details">
      <div className="user-info">
        <p>{props.user.first_name} {props.user.first_name}</p> </div>
      <button
        className="button button--link btn-delete"
        onClick={(e) => {
          props.onDelete(e, props.user.id);
        }}
      >
        Delete
      </button>
    </div>
  </li>
);

UserInfo.PropTypes = {
  user: PropTypes.object.isRequired,
  onClick: PropTypes.func
}

export default UserInfo;
