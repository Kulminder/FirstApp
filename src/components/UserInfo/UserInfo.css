.person-item{
    background: #fff;
    margin: 18px 19px 55px 18px;
    float: left;
    border: 2px solid #ddd;
    width: 212px;
    text-align: center;
}
.avatar-img{
  width: 80px;
  height: 80px;
  border-radius: 50%;
  margin-top: -40px;
  position: relative;
  z-index: 9;
}
.user-name{
  font-size: 18px;
  color: #E93939;
}
.button-link{
  font-size: 14px;
  color: #007FA3;
  cursor: pointer;
  text-align: left;
  margin: 10px 0;
  float: left;
}
.person-details{
  background: #fff;
  padding:10px;
}

