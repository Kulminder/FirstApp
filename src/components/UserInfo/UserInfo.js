import React from 'react';
import './UserInfo.css';
import propTypes from 'prop-types'


const UserInfo = (props) => props.users.map((user, index) => {

  return (
    <div key={user.id} className="person-item">
      <img alt={user.first_name} className="avatar-img" src={user.avatar} />
      <div className="person-details">
        <div className="user-name">
          {user.first_name} {user.first_name}
        </div>
        <a className="button-link" onClick={(e) => { props.onDelete(e, index); }}
        >Delete</a>
      </div>
    </div>
  )
})

UserInfo.defaultProps = {
  users: []
}

UserInfo.propTypes = {
  users: propTypes.array.isRequired,
  onClick: propTypes.func
}

export default UserInfo;
