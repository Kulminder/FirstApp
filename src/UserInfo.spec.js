import React from "react";
import { shallow } from 'enzyme';
import UserInfo from "./UserInfo";
import toJson from 'enzyme-to-json';
import sinon from 'sinon';

const props = {
  user: [
    {
      id: 5,
      first_name: "Charles",
      last_name: "Morris",
      avatar: "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
    }
  ]
};

beforeEach(() => {
  sinon.stub(console, 'error');
});

afterEach(() => {
  console.error.restore();
});

describe("UserInfo component", () => {
  it('should renders correctly', () => {
    const wrapper = shallow(
      <UserInfo user={props} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });


  it('should invoke pleaseMockMe callback', () => {
    let onDelete = jest.fn();
    let wrapper = shallow(<UserInfo user={props} onDelete={onDelete} />);
    wrapper.find('button').simulate('click', { target: { value: 1 } }); 
    expect(onDelete).toHaveBeenCalled();
  });

});
