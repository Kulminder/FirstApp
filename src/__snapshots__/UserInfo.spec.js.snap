// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component should renders correctly 1`] = `
<li
  className="person-list"
>
  <img
    className="avatar-img"
  />
  <div
    className="person-details"
  >
    <div
      className="user-info"
    >
      <p>
         
      </p>
       
    </div>
    <button
      className="button button--link btn-delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</li>
`;
